src/documents/index.py

```diff
@@ -368,13 +370,47 @@ def _get_query(self):
         return q, mask
 
 
-def autocomplete(ix, term, limit=10):
-    with ix.reader() as reader:
-        terms = []
-        for score, t in reader.most_distinctive_terms(
-            "content",
-            number=limit,
-            prefix=term.lower(),
-        ):
-            terms.append(t)
-        return terms
+def autocomplete(ix, term, limit=10, user_id=None, is_superuser=False):
+    """
+    Mimics whoosh.reading.IndexReader.most_distinctive_terms, including permissions
+    """
+    terms = []
+
+    with ix.searcher(weighting=TF_IDF()) as s:
+        qp = QueryParser("content", schema=ix.schema)
+        q = qp.parse(f"{term.lower()}*")
+        user_criterias = get_permissions_criterias(user_id, is_superuser)
+
+        results = s.search(
+            q,
+            terms=True,
+            filter=query.Or(user_criterias) if user_criterias is not None else None,
+        )
+
+        termCounts = dict()
```

Comment:
This looks a lot like a [Counter](https://docs.python.org/3/library/collections.html#collections.Counter).  No need to check if the term is in the structure, `c["hello"] += 1` will create if needed and increment.

Reply:
Awesome thanks, this definitely helps make this cleaner!